WooCommerce/Classes/POS/Controllers/PointOfSaleItemsController.swift

```diff
@@ -79,12 +79,13 @@ class PointOfSaleItemsController: PointOfSaleItemsControllerProtocol {
     /// - Returns: A boolean that indicates whether there is next page for the paginated items.
     @MainActor
     private func fetchItems(pageNumber: Int) async throws -> Bool {
-        let (newItems, hasNextPage) = try await itemProvider.providePointOfSaleItems(pageNumber: pageNumber)
+        let pagedItems = try await itemProvider.providePointOfSaleItems(pageNumber: pageNumber)
+        let newItems = pagedItems.items
         let uniqueNewItems = newItems.filter { newItem in
             !allItems.contains(newItem)
         }
         allItems.append(contentsOf: uniqueNewItems)
-        return hasNextPage
+        return pagedItems.hasMorePages
```

Comment:
It's been there a while now... but this function is feeling a bit strange now.

It's called `fetchItems`, which it does, but it doesn't return them, it returns a Bool.

If I didn't have the doc comment, I'd expect that to indicate success/failure of the fetch, not whether there are more pages. It's not a huge deal on this private function but it's niggling at me a bit.

We can leave it for now, this one will change more when the state model changes, but it'd be good to know your thoughts. I guess just returning the whole `PagedItems` actually would make more sense...